app.js

```diff
@@ -71,12 +71,52 @@ app.patch('/api/v1/projects/:id', async (request, response) => {
 
   try {
     const patchedProject = await database('projects').where('id', id).update({title: newProject.title}, ['id', 'title']);
-
     if (patchedProject.length) {
       response.status(201).json({id: patchedProject[0].id, title: patchedProject[0].title});
     } else {
       response.status(404).json({
-        error: `Could not find a project with id: ${request.params.id}`
+        error: `Could not find a project with id: ${id}`
+      });
+    }
+  } catch (error) {
+    response.status(500).json({error: 'internal server error' })
+  }
+});
+
+app.patch('/api/v1/palettes/:id', async (request, response) => {
+  const { id } = request.params;
+  const newPalette = request.body;
+  try {
+    const patchedPalette = await database('palettes').where('id', id)
+    .update(
```

Comment:
I haven't worked with a PATCH yet, but it looks like knex has commands for everything and makes it easy. This is nice

Reply:
You're the man